pkg/stanza/fileconsumer/internal/tracker/tracker_other.go

```diff
@@ -3,23 +3,24 @@
 
 //go:build !windows
 
-package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+package tracker // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/tracker"
 
 import (
 	"context"
 	"sync"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fileset"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
 )
 
 // Take care of files which disappeared from the pattern since the last poll cycle
 // this can mean either files which were removed, or rotated into a name not matching the pattern
 // we do this before reading existing files to ensure we emit older log lines before newer ones
-func (m *Manager) preConsume(ctx context.Context) {
-	lostReaders := make([]*reader.Reader, 0, m.previousPollFiles.Len())
+func (t *Tracker) PreConsume(ctx context.Context) {
```

Comment:
Similar to other comments, I think the notion of consumption should remain the concern of the manager. However, the tracker can support it with a function that detects and returns the list of lost files. e.g. `t.LostFiles()`